klogr/klogr.go

```diff
@@ -255,4 +260,11 @@ func (l klogger) WithValues(kvList ...interface{}) logr.Logger {
 	return new
 }
 
+func (l klogger) WithCallDepth(depth int) logr.Logger {
+	new := l.clone()
+	new.callDepth += depth
+	return new
+}
+
 var _ logr.Logger = klogger{}
+var _ logr.CallDepthLogger = klogger{}
```

Comment:
Usage of the `CallDepthLogger` type in this compile-time check is the only reason why this PR updates logr to master.

We can either:
- wait for a new tagged logr release and use that or
- omit this check here and leave the logr dependency unchanged.